pdf/core/parser.go

```diff
@@ -53,6 +53,9 @@ type PdfParser struct {
 	// the length reference (if not object) prior to reading the actual stream.  This has risks of endless looping.
 	// Tracking is necessary to avoid recursive loops.
 	streamLengthReferenceLookupInProgress map[int64]bool
+
+	// Hack to allow parsing of PDF fragments without reporting an EOF error.
+	suppressEOF bool
```

Comment:
Can we handle the EOF within the tests?  If we want to provide the functionality it should be more generic than for this specific use case.  The other option is to leave here, but then we can describe what it does rather than pointing to the specific case of PDF fragments.

Reply:
e.g. "suppressEOF suppresses io.EOF errors such that the errors are logged and discarded rather than passing up the error return stack."